### PR TITLE
Travis Continuous Integration

### PR DESCRIPTION
**Pull request by danielpygo**
- Adding .travis.yml
- Adding run_test.sh

### CODE REVIEW

**Comment by danielpygo** [comment]
https://travis-ci.org/danielpygo/kubeplus/builds/508112699

Right now, my run_test.sh script does work locally, but when I run it on travis it fails to deploy the operator:
https://travis-ci.org/danielpygo/kubeplus/builds/508112699
Perhaps a helm issue, I need to print out the logs of the kubeplus deployer to debug further

File: scripts/run_test.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+echo "Successfully deployed ..."
+echo ""
+echo "Creating postgres instance ..."
+kubectl create -f $PROJECT_HOME/examples/postgres/postgres1.yaml
+until kubectl get pods -lapp=postgres1 -o jsonpath="$JSONPATH" 2>&1 | grep -q "Ready=True"; do sleep 3;echo "Waiting for postgres instance to be available"; kubectl get pods -n default; done
+echo "Successfully created postgres instance ..."
+echo ""
+echo "Deleting postgres instance ..."
+kubectl delete -f $PROJECT_HOME/examples/postgres/postgres1.yaml
+echo "Successfully deleted postgres instance ..."
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: Lets add a check for 'kubectl delete', similar to 'kubectl create'.

File: scripts/run_test.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+echo ""
+echo "Deleting postgres instance ..."
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: This should be echo "Deleting postgres operator ..."

File: scripts/run_test.sh [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    echo "Creating postgres instance ..."
+    kubectl create -f $PROJECT_HOME/examples/postgres/postgres1.yaml
+    timeout=600
+    sleep_time=3
+    loops=$((timeout/sleep_time))
+    count=0
+    result=$(kubectl -n default get pods -lapp=postgres1  -o jsonpath="$JSONPATH" 2>&1 | grep "Ready=True")
+    while ! [[ "$result" ]]; do
+        if [[ "$count" -gt $loops ]]; then
+            exit 1
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: Can you print postgres pods here -- similar to how you are doing it create_operator (lines 80-87)?

**Comment by devdattakulkarni** [comment]
This is looking really good. Just one comment about displaying logs of postgres pod when create_postgres() fails or times out.

File: scripts/run_test.sh (hunk 3 above)
Comment (danielpygo) [reply scripts/run_test.sh]: Sure

**Comment by danielpygo** [comment]
Should be ready .

File: scripts/run_test.sh (hunk 3 above)
Comment (devdattakulkarni) [reply scripts/run_test.sh]: This is not correct.

You should do following:

pod=`kubectl get pods | grep -v postgres-operator | grep postgres | awk '{print $1}'`

kubectl logs $pod

File: scripts/run_test.sh [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    loops=$((timeout/sleep_time))
+    count=0
+    result=$(kubectl -n default get pods -lapp=postgres-operator  -o jsonpath="$JSONPATH" 2>&1 | grep "Ready=True")
+    pod=`kubectl get pods | grep kubeplus | awk '{print $1'}`
+    while ! [[ "$result" ]]; do
+        if [[ "$count" -gt $loops ]]; then
+            echo "Unable to create a postgres operator."
+            echo ""
+            echo "Operator deployer logs: "
+            kubectl logs $pod operator-deployer
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: This should be:

kubectl logs $pod -c operator-deployer

File: scripts/run_test.sh [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+            echo ""
+            echo "Operator manager logs: "
+            kubectl logs $pod operator-manager
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: This should be:

kubectl logs $pod -c operator-manager

File: scripts/run_test.sh [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+}
+function create_postgres(){
+    echo "Creating postgres instance ..."
+    kubectl create -f $PROJECT_HOME/examples/postgres/postgres1.yaml
+    timeout=600
+    sleep_time=3
+    loops=$((timeout/sleep_time))
+    count=0
+    result=$(kubectl -n default get pods -lapp=postgres1  -o jsonpath="$JSONPATH" 2>&1 | grep "Ready=True")
+    pod=`kubectl get pods | grep kubeplus | awk '{print $1'}`
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: pod=`kubectl get pods | grep -v postgres-operator | grep postgres | awk '{print $1}'`

File: scripts/run_test.sh [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+    while ! [[ "$result" ]]; do
+        if [[ "$count" -gt $loops ]]; then
+            echo "Unable to create postgres."
+            echo ""
+            echo "Operator deployer logs: "
+            kubectl logs $pod -c operator-deployer
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: kubectl logs $pod (only this and no '-c operator-deployer')

File: scripts/run_test.sh [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+            echo ""
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: No need of this line.

File: scripts/run_test.sh [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+            echo "Operator manager logs: "
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: No need of this line.

File: scripts/run_test.sh [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+            kubectl logs $pod -c operator-manager
```
Comment (devdattakulkarni) [inline scripts/run_test.sh]: No need of this line.

**Pull request merged**